SHARE-479 PUT /notification/{id}/read

Kropiunig opened the pull request:
-implemented /notification/{id}/read API
-removed old API
-wrote behat tests for API

---
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [x] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no warnings and errors 
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Verify that all tests are passing (CI), if not please state the test cases in the [section](#Tests) below
- [x] Perform a self-review of the changes
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [x] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer
- [x] Check that your pull request has been successfully deployed to https://web-test-1.catrob.at/

Kropiunig committed:
SHARE-479 PUT /notification/{id}/read

-implemented notification/count API
-removed old API
-replaced usages of old API with new API
-wrote test for API

Kropiunig committed:
SHARE-479 PUT /notification/{id}/read

-implemented /notification/{id}/read API
-removed old API
-wrote behat tests for API

dmetzner commented on src/Api/NotificationsApi.php:
```diff
@@ -21,9 +21,14 @@ public function notificationIdReadPut(int $id, ?string $accept_language = null,
-    // TODO: Implement notificationIdReadPut() method.
+    $user = $this->facade->getAuthenticationManager()->getUserFromAuthenticationToken($this->getAuthenticationToken());
+    if (is_null($user)) {
+      $responseCode = Response::HTTP_FORBIDDEN;
```
401

dmetzner replied:
"-..anager()->getAuthenticatedUser() sollte auch reichen

dmetzner commented on src/Api/Services/Notifications/NotificationsApiProcessor.php:
```diff
@@ -3,7 +3,32 @@
+    }
+    $this->notification_manager->markSeen([$notification_seen]);
+
+    return Response::HTTP_NO_CONTENT;
```
The DB processor must not be aware of the HTTP status code. Just return a bool true/false if the operation was successful.

dmetzner commented on src/Application/Controller/User/NotificationsController.php:
```diff
@@ -24,7 +24,7 @@ class NotificationsController extends AbstractController
 
   /**
-   * @Route("/user_notifications", name="notifications", methods={"GET"})
+   * @Route("/user_notifications", name="notifications", methods={"GET"})SHARE-479
```
typo?

dmetzner commented on src/DB/EntityRepository/User/Notification/NotificationRepository.php:
```diff
@@ -76,4 +76,15 @@ public function getFollowNotificationForUser(User $owner = null, User $follow_fr
+
+  public function getNotificationByID(int $id): ?CatroNotification
+  {
+    $query_builder = $this->createQueryBuilder('e');
```
Nothing wrong with the query. Just FYI every entity repository should already have a 'find' method which searches an entity by its ID

`    public function find($id, $lockMode = null, $lockVersion = null)
`

dmetzner commented on tests/PhpUnit/Api/NotificationsApiTest.php:
```diff
@@ -93,7 +93,7 @@ public function testNotificationIdReadPut(): void
     $response = $this->object->notificationIdReadPut(1, null, $response_code, $response_headers);
 
-    $this->assertEquals(Response::HTTP_NOT_IMPLEMENTED, $response_code);
+    $this->assertEquals(Response::HTTP_FORBIDDEN, $response_code);
```
401

dmetzner commented on src/Api/Services/Notifications/NotificationsApiLoader.php:
```diff
@@ -3,14 +3,26 @@
+
+  public function findNotificationByID(string $id): ?CatroNotification
+  {
+    return $this->notification_repository->getNotificationByID($id);
```
phpstan*

dmetzner reviewed (changes requested):
Please fix the minor issues and merge conflicts - All in all the PR looks fine!

Kropiunig committed:
SHARE-479 PUT /notification/{id}/read

-implemented /notification/{id}/read API
-removed old API
-wrote behat tests for API

Kropiunig committed:
SHARE-479 PUT /notification/{id}/read

-implemented /notification/{id}/read API
-removed old API
-wrote behat tests for API

Kropiunig committed:
SHARE-479 PUT /notification/{id}/read

-implemented /notification/{id}/read API
-removed old API
-wrote behat tests for API

Kropiunig committed:
SHARE-479 PUT /notification/{id}/read

-implemented /notification/{id}/read API
-removed old API
-wrote behat tests for API

Kropiunig committed:
SHARE-479 PUT /notification/{id}/read

-implemented /notification/{id}/read API
-removed old API
-wrote behat tests for API
